src/app.js

```
@@ -43,18 +44,50 @@
+      } else {
+        countdownEl.textContent = '活动已结束';
+      }
+    };
+    updateCountdown();
+  }
+
   renderProducts() {
     const productsContainer = document.getElementById('products');
+    productsContainer.innerHTML = '';
     products.forEach(product => {
+      const discounted = product.discount && product.discount < 1;
```

ESLint

[error] Extra semicolon.

(semi)